### PR TITLE
Bugfix/prevent disabled button click

### LINKED ISSUE #1551: Axa button submits form even when disabled
Axa button of type submit will submit form even if disabled.

Expected behavior:
Disabled button will not trigger either submit or click events

Reproduction link
https://codesandbox.io/embed/pedantic-mirzakhani-wcn0t?fontsize=14&hidenavigation=1&theme=dark

### PR DESCRIPTION
**Pull request by 01231**
Fixes #1551.

# Done is when (DoD):
- [x] I have added UI/Unit tests for my changes
- [ ] I added modified component properties to the typescript typings
- [ ] Design has been reviewed with (here name of reviewer)
- [x] My changes are accordingly to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)
- [x] Old tests and eslint rule are not broken
- [ ] My changes are well documented in the README and showcased in the stories 
- [x] Tested with IE

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas (focus on **WHY**)
- [ ] I have made/updated the ChangeLog Section in the README

### CODE REVIEW

File: src/components/10-atoms/button/index.js [hunk 1]
```diff
@@ -28,6 +30,12 @@ class AXAButton extends LitElement {
     `;
   }
 
+  // Parent class InlineStyles needs a static method to retrive styles
+  // name of such method is passed when calling: this.inlineStyles('resetHeadingCss');
+  static get resetHeadingCss() {
```
Comment (LucaMele) [inline src/components/10-atoms/button/index.js]: @NoahLiechti Please rename this as it does not reset headings rather is blocks mouse events or you can also call it direct child styles

File: src/components/10-atoms/button/index.js [hunk 2]
```diff
@@ -59,6 +67,8 @@ class AXAButton extends LitElement {
   }
 
   firstUpdated() {
+    this.inlineStyles('resetHeadingCss');
```
Comment (LucaMele) [inline src/components/10-atoms/button/index.js]: dtto

**Review by LucaMele — CHANGES_REQUESTED** [review]

**Review by raphaellueckl — DISMISSED** [review]
very nice

**Review by LucaMele — APPROVED** [review]
very nice

**Pull request merged**